tests.c

```diff
@@ -450,6 +450,65 @@ test_opcode_0xc6(void)
   cpu_write_mem(&cpu, 0x0001, 0x00);
 }
 
+void
+test_opcode_0xc8(void)
+{ // RZ
+  i8080 cpu;
+  cpu_init(&cpu);
+
+  // case where zero flag not set
+  cpu.pc = 0x4341;                                  // NOLINT
+  update_zero_flag(&cpu, 1);                        // had a remainder
+  int code_found = execute_instruction(&cpu, 0xc8); // NOLINT
+  CU_ASSERT(code_found >= 0);
+  CU_ASSERT(cpu.pc == 0x4342); // NOLINT
+
+  // case where zero flag set
+  cpu_write_mem(&cpu, 0x0001, 0xAA); // NOLINT
+  cpu_write_mem(&cpu, 0x0002, 0xBB); // NOLINT
+  cpu.sp = 0x0001;
+  update_zero_flag(&cpu, 0); // no remainder
+
+  code_found = execute_instruction(&cpu, 0xc8); // NOLINT
+
+  CU_ASSERT(code_found >= 0);
+  CU_ASSERT(cpu.pc == 0xBBAA); // NOLINT
+  CU_ASSERT(cpu.sp == 0x0003); // NOLINT
+
+  // clean up
+  cpu_write_mem(&cpu, 0x0001, 0x00); // NOLINT
+  cpu_write_mem(&cpu, 0x0002, 0x00); // NOLINT
+}
+
+void
+test_opcode_0xca(void)
+{ // JZ
+  i8080 cpu;
+  cpu_init(&cpu);
+
+  // case where zero flag not set
+  cpu.pc = 0x4341;                                  // NOLINT
+  cpu_write_mem(&cpu, cpu.pc + 1, 0x28);            // NOLINT
+  cpu_write_mem(&cpu, cpu.pc + 2, 0xb7);            // NOLINT
+  update_zero_flag(&cpu, 1);                        // had a remainder
+  int code_found = execute_instruction(&cpu, 0xca); // NOLINT
+  CU_ASSERT(code_found >= 0);
+  CU_ASSERT(cpu.pc == 0x4344); // NOLINT
+
+  // case where zero flag set
+  cpu.pc = 0x5441;                              // NOLINT
+  cpu_write_mem(&cpu, cpu.pc + 1, 0x3a);        // NOLINT
+  cpu_write_mem(&cpu, cpu.pc + 2, 0xa9);        // NOLINT
+  update_zero_flag(&cpu, 0);                    // no remainder
+  code_found = execute_instruction(&cpu, 0xca); // NOLINT
+  CU_ASSERT(code_found >= 0);
+  CU_ASSERT(cpu.pc == 0xa93a); // NOLINT
+
+  // cleanup
+  cpu_write_mem(&cpu, 0x0001, 0x00); // NOLINT
+  cpu_write_mem(&cpu, 0x0002, 0x00); // NOLINT
```

Comment:
Cleaning up wrong memory - I think this is fixed in the other PR?

Reply:
That's correct.